modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -149,18 +179,134 @@ def _perform_node_fail_scenario(self, test_config):
 
             start_load_app(self.test_context, ignite_config=load_config, params=params, modules=modules)
 
-        data = simulate_nodes_failure(servers, failed_nodes, survived_node)
+        for node in failed_nodes:
+            self.logger.info(
+                "Simulating failure of node '%s' (order %d) on '%s'" % (node_id(node), order(node), node.name))
+
+        data = self._simulate_nodes_failure(servers, node_fail_task(ignite_config, test_config), failed_nodes,
```

Comment:
Node stopping relies on parallel and simulatous execution of configurable task. The parameters are outside of this execution’s API and are not it’s responsibilities. But task relies on the configuration. It’s much more simpler to form the task somewhere by a method and return lambda. Also helps to keep admissible function parameters.

Reply:
Also shortens makes more reeadable, simplifies `_simulate_nodes_failure`.

Reply:
Not sure I got an answer.
node_fail_task content (or call) can be (should?) inlined to _simulate_nodes_failure instead of being its param since it can be used only inside this method.

Reply:
_simulate_nodes_failure will look like `def _simulate_nodes_failure(self, servers, failed_nodes, survived_node, test_config, ignite_config):
        """
        Perform node failure scenario
        """
        kill_node_task = node_fail_task(ignite_config, test_config)` 
And pylint will say `Too many arguments (6/5) (too-many-arguments)` and `Too many local variables (17/15) (too-many-locals)`

Reply:
_simulate_nodes_failure should not be overloaded any more with any locals and parameters.

Reply:
got it